src/reducers/db.js

```diff
@@ -32,6 +33,24 @@ export const dbReducer = (state = {}, action) => {
         };
       }
       break;
+    case OPTIMISTIC_UPDATE_PHONE:
+      const location = state[action.payload.locationId];
+      const idx = location.Phones.map( phone => phone.id).indexOf(action.payload.phoneId)
+      const phone = location.Phones[idx]
+      const newPhones = location.Phones.slice(0,idx).concat({...phone, ...action.payload.params}).concat(location.Phones.slice(idx+1))
```

Comment:
I think this could be:
```
const newPhones = [...location.Phones.slice(0, idx), {...phone, ...action.payload.params}, ...location.Phones.slice(idx+1)];
```

Reply:
Fixed in cdf652af033e7b9e18f52372d76b55de2ae0efb7